### DISCUSSION TITLE
ModuleNotFoundError: No module named 'httpx'

### DISCUSSION THREAD

**Discussion by PECgeo** [searxng/searxng#2496] [Q&A] [answered]
Good Day I am not sure if this is a bug or I screwed up somewhere so I thought I would try here. This is a fresh install in a FreeBSD jail. 
I am not sure why "python searx/webapp.py" is not picking up httpx. It is listed in /home/searxng/searx-pyenv/lib/python3.9/site-packages/httpx/ and listed in packages installed. 
Any help or suggestion would be appreciated
Thanks
Geordie

uname -a
FreeBSD searxng.lan 13.2-RELEASE FreeBSD 13.2-RELEASE releng/13.2-n254617-525ecfdad597 GENERIC amd64

root@searxng:/ # su searxng
[searxng@searxng /]$ source /home/searxng/searx-pyenv/bin/activate
(searx-pyenv) [searxng@searxng /]$ cd /usr/home/searxng/searxng-src
(searx-pyenv) [searxng@searxng /usr/home/searxng/searxng-src]$ pip install -e /home/searxng/searxng-src
Obtaining file:///home/searxng/searxng-src
  Preparing metadata (setup.py) ... done
Requirement already satisfied: certifi==2023.5.7 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (2023.5.7)
Requirement already satisfied: babel==2.12.1 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (2.12.1)
Requirement already satisfied: flask-babel==3.1.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (3.1.0)
Requirement already satisfied: flask==2.3.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (2.3.2)
Requirement already satisfied: jinja2==3.1.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (3.1.2)
Requirement already satisfied: lxml==4.9.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (4.9.2)
Requirement already satisfied: pygments==2.15.1 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (2.15.1)
Requirement already satisfied: python-dateutil==2.8.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (2.8.2)
Requirement already satisfied: pyyaml==6.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (6.0)
Requirement already satisfied: httpx[http2]==0.21.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (0.21.2)
Requirement already satisfied: Brotli==1.0.9 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (1.0.9)
Requirement already satisfied: uvloop==0.17.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (0.17.0)
Requirement already satisfied: httpx-socks[asyncio]==0.7.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (0.7.2)
Requirement already satisfied: setproctitle==1.3.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (1.3.2)
Requirement already satisfied: redis==4.5.5 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (4.5.5)
Requirement already satisfied: markdown-it-py==2.2.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (2.2.0)
Requirement already satisfied: typing_extensions==4.6.3 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (4.6.3)
Requirement already satisfied: fasttext-predict==0.9.2.1 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (0.9.2.1)
Requirement already satisfied: pytomlpp==1.0.13 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from searxng==2023.6.10+5f11155c) (1.0.13)
Requirement already satisfied: Werkzeug>=2.3.3 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from flask==2.3.2->searxng==2023.6.10+5f11155c) (2.3.6)
Requirement already satisfied: itsdangerous>=2.1.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from flask==2.3.2->searxng==2023.6.10+5f11155c) (2.1.2)
Requirement already satisfied: click>=8.1.3 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from flask==2.3.2->searxng==2023.6.10+5f11155c) (8.1.3)
Requirement already satisfied: blinker>=1.6.2 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from flask==2.3.2->searxng==2023.6.10+5f11155c) (1.6.2)
Requirement already satisfied: importlib-metadata>=3.6.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from flask==2.3.2->searxng==2023.6.10+5f11155c) (6.6.0)
Requirement already satisfied: pytz>=2022.7 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from flask-babel==3.1.0->searxng==2023.6.10+5f11155c) (2023.3)
Requirement already satisfied: httpcore<0.15.0,>=0.14.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (0.14.7)
Requirement already satisfied: python-socks>=2.0.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (2.3.0)
Requirement already satisfied: async-timeout>=3.0.1 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (4.0.2)
Requirement already satisfied: charset-normalizer in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx[http2]==0.21.2->searxng==2023.6.10+5f11155c) (3.1.0)
Requirement already satisfied: sniffio in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx[http2]==0.21.2->searxng==2023.6.10+5f11155c) (1.3.0)
Requirement already satisfied: rfc3986[idna2008]<2,>=1.3 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx[http2]==0.21.2->searxng==2023.6.10+5f11155c) (1.5.0)
Requirement already satisfied: h2<5,>=3 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpx[http2]==0.21.2->searxng==2023.6.10+5f11155c) (4.1.0)
Requirement already satisfied: MarkupSafe>=2.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from jinja2==3.1.2->searxng==2023.6.10+5f11155c) (2.1.3)
Requirement already satisfied: mdurl~=0.1 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from markdown-it-py==2.2.0->searxng==2023.6.10+5f11155c) (0.1.2)
Requirement already satisfied: six>=1.5 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from python-dateutil==2.8.2->searxng==2023.6.10+5f11155c) (1.16.0)
Requirement already satisfied: hyperframe<7,>=6.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from h2<5,>=3->httpx[http2]==0.21.2->searxng==2023.6.10+5f11155c) (6.0.1)
Requirement already satisfied: hpack<5,>=4.0 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from h2<5,>=3->httpx[http2]==0.21.2->searxng==2023.6.10+5f11155c) (4.0.0)
Requirement already satisfied: h11<0.13,>=0.11 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpcore<0.15.0,>=0.14.0->httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (0.12.0)
Requirement already satisfied: anyio==3.* in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from httpcore<0.15.0,>=0.14.0->httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (3.7.0)
Requirement already satisfied: idna>=2.8 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from anyio==3.*->httpcore<0.15.0,>=0.14.0->httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (3.4)
Requirement already satisfied: exceptiongroup in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from anyio==3.*->httpcore<0.15.0,>=0.14.0->httpx-socks[asyncio]==0.7.2->searxng==2023.6.10+5f11155c) (1.1.1)
Requirement already satisfied: zipp>=0.5 in /usr/home/searxng/searx-pyenv/lib/python3.9/site-packages (from importlib-metadata>=3.6.0->flask==2.3.2->searxng==2023.6.10+5f11155c) (3.15.0)
Installing collected packages: searxng
  Attempting uninstall: searxng
    Found existing installation: searxng 2023.6.10+5f11155c
    Uninstalling searxng-2023.6.10+5f11155c:
      Successfully uninstalled searxng-2023.6.10+5f11155c
  Running setup.py develop for searxng
Successfully installed searxng-2023.6.10+5f11155c

root@searxng:/ # su searxng
[searxng@searxng /]$ cd /usr/home/searxng/searxng-src
[searxng@searxng /usr/home/searxng/searxng-src]$ export SEARXNG_SETTINGS_PATH="/usr/local/etc/searx/settings.yml"
[searxng@searxng /usr/home/searxng/searxng-src]$ python searx/webapp.py
Traceback (most recent call last):
  File "/usr/home/searxng/searxng-src/searx/webapp.py", line 27, in <module>
    import httpx
ModuleNotFoundError: No module named 'httpx'

**Comment by return42** [comment, marked as answer]
SearXNG can't installed by pip .. please follow the [docs](https://docs.searxng.org/admin/installation.html)

We do not have a installation instruction or a script for BSD / its untested:

- https://github.com/searxng/searxng/issues/1418
